### PR TITLE
Cleanup deprecated query options

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
- Cleanup the deprecated debug options, and old PQL related query options
- Fix the double query parsing in multi-stage engine
- Fix the inconsistent behavior when query parsing fails: do not log or emit metric because it is pure user error

### CODE REVIEW

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java [hunk 1]
```diff
@@ -85,24 +84,22 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
...
-    if (request.has(CommonConstants.Broker.Request.QUERY_OPTIONS)) {
-      sqlNodeAndOptions.setExtraOptions(
-          RequestUtils.getOptionsFromJson(request, CommonConstants.Broker.Request.QUERY_OPTIONS));
-    }
 
     if (_multiStageBrokerRequestHandler != null && Boolean.parseBoolean(
-        sqlNodeAndOptions.getOptions().get(CommonConstants.Broker.Request.QueryOptionKey.USE_MULTISTAGE_ENGINE))) {
-      return _multiStageBrokerRequestHandler.handleRequest(request, requesterIdentity, requestContext, httpHeaders);
+        sqlNodeAndOptions.getOptions().get(Request.QueryOptionKey.USE_MULTISTAGE_ENGINE))) {
+      return _multiStageBrokerRequestHandler.handleRequest(request, sqlNodeAndOptions, requesterIdentity,
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java:101]: This is an important fix. Currently we are not passing the parsed SQL into the multi-stage engine, which results in parsing the query twice

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java [hunk 2]
```diff
@@ -85,24 +84,22 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
...
     requestContext.setBrokerId(_brokerId);
+
+    // Parse the query if needed
     if (sqlNodeAndOptions == null) {
       try {
-        sqlNodeAndOptions = RequestUtils.parseQuery(request.get(CommonConstants.Broker.Request.SQL).asText(), request);
+        sqlNodeAndOptions = RequestUtils.parseQuery(request.get(Request.SQL).asText(), request);
       } catch (Exception e) {
-        LOGGER.info("Caught exception while compiling SQL: {}, {}", request, e.getMessage());
-        _brokerMetrics.addMeteredGlobalValue(BrokerMeter.REQUEST_COMPILATION_EXCEPTIONS, 1);
```
Comment (gortiz) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java:93]: If we don't increase it here, where should we increase `REQUEST_COMPILATION_EXCEPTIONS`? Should we delete it?

**Review by gortiz — APPROVED** [review]

**Comment by gortiz** [comment]
LGTM, but there are multiple tests failing, so I guess we would need to fix something else

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BrokerRequestHandlerDelegate.java:93]: We count query planning phase (in v1 the step of converting SqlNode to PinotQuery) as request compilation

**Comment by Jackie-Jiang** [comment]
> LGTM, but there are multiple tests failing, so I guess we would need to fix something else

Find the reason. Currently we only set query option when it is not empty, but the following code assume it is always set. After removing the deprecated options it breaks. Changed the code to always set the query option even if it is empty.

**Pull request merged**